VotingApplication/VotingApplication.Web/Views/Routes/Manage.cshtml

```diff
@@ -9,7 +9,18 @@
                 <h4 class="icon-padded-left">Question:</h4>
             </div>
             <div>
-                <span class="icon-padded-left"><input class="manage-name-input" ng-model="poll.Name" required ng-model-options="{debounce: 1000}" ng-change="updatePoll()" /></span>
+                <form ng-submit="toggleNameEdit=false;updatePoll()">
+                    <span class="icon-padded-left">
+                        <span ng-if="!toggleNameEdit">
+                            {{poll.Name}}
+                            <span class="pull-right"><a class="clickable" ng-click="$parent.toggleNameEdit=true">Edit</a></span>
+                        </span>
+                        <span ng-if="toggleNameEdit">
+                            <input class="manage-name-input" ng-model="poll.Name" required />
+                            <span class="glyphicon glyphicon-ok pull-right clickable form-aligned-top" ng-click="$parent.toggleNameEdit=false;$parent.updatePoll()"></span>
```

Comment:
Ctrl+Z is a tech-savvy shortcut, so we shouldn't rely on it. We also shouldn't rely on people remembering what the original name was.